File: Electronics_Shop_17.Services/Database/Notification.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class Notification : IHelperId
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Message { get; set; }
        public DateTime DateCreated { get; set; }
        public bool IsGeneral { get; set; }

        public virtual List<UserNotification> UserNotifications { get; set; }
    }
}

File: angular17_web/src/app/models/notification/notification.ts
export class Notification {
  id!: number;
  title!: string;
  message!: string;
  dateCreated!: Date;
  isGeneral!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.title = data.title;
    this.message = data.message;
    this.dateCreated = data.dateCreated ? new Date(data.dateCreated) : undefined as any;
    this.isGeneral = data.isGeneral;
  }
}
